### PR TITLE
Feature/add obs source

### PR DESCRIPTION
**Pull request by MarcelCaron-NOAA**
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

This PR includes the following changes:
- _Feature:_ Add the name of the observation data source used to produce the statistics featured on each graphic.
- _Bugfix:_ An EVS script that prunes stat files into a manageable size for plotting uses subprocess.Popen to reading files.  This method launches background processes that aren't killed when the PBS job is killed.  The bugfix uses subprocess.run instead.  _Fix outcome:_ When a PBS job is killed, all background processes are killed too (this was already demonstrated when the same change was applied to the global_det component).
- _Bugfix:_ Snowfall plots are being backed up to restart directories underneath a "precip" subdirectory.  This technically works, but is confusing because "precip" also matches a verif_case for EVS-cam that does not involve snowfall jobs.  That "precip" subdirectory was changed to "snowfall".  _Fix outcome:_ The snowfall plots are now stored in restart subdirectories that more closely follow EVS naming convention.

## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by?

No
* Do you have any planned upcoming annual leave/PTO?

Yes, July 10 - August 16 2025
* Are there any changes needed in the times when the jobs are supposed to run/kick-off?

No
  
- [x] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [x] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [x] References the feature branch for `HOMEevs` are removed from the code.
- [x] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [x] Jobs over 15 minutes in runtime have restart capability.
- [x] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [x] Jobs contain the appropriate file checking and don't run METplus for any missing data.
- [x] Code is using METplus wrappers structure and not calling MET executables directly.
- [x] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

#### Set up jobs

- symlink the EVS_fix directory locally as "fix"
- In each driver script for the listed jobs, edit the following environment variables:
**HOMEevs** - set to your test EVS directory
**COMIN** - when needed, set to `/lfs/h2/emc/vpppg/noscrub/emc.vpppg/${NET}/$evs_ver_2d`
**COMOUT** - set to your test output directory
**KEEPDATA** - (_optional_) set to `"YES"`
**DATAROOT** - (_optional_) set to your test DATAROOT directory

#### Running jobs

I recommend testing the following jobs:
```
jevs_cam_headline_plots
jevs_cam_precip_plots_last31days
jevs_cam_snowfall_plots
```

[Total: 3 plots jobs]

#### Testing jobs

_**plots:**_
- **jevs_cam_snowfall_plots**: submit using `qsub -v vhr=00,VDATE=20250131 $driver`
- **All other plots**: submit using `qsub -v vhr=00 $driver`

#### Checking jobs
- Log files should be checked by the developer for the following keywords:
```
check="FATAL\|WARNING\|error\|Error\|Killed\|Cgroup\|argument expected\|No such file\|cannot\|failed\|unexpected\|exceeded"
grep "$check" $logfile
```

### CODE REVIEW

**Comment by ShannonShields-NOAA** [comment]
Hi @MarcelCaron-NOAA Can you please sync your branch before I pull it?

**Comment by ShannonShields-NOAA** [comment]
Thanks! I will kick off the jobs shortly.

**Comment by MarcelCaron-NOAA** [comment]
Synced!  Thanks @ShannonShields-NOAA

**Comment by ShannonShields-NOAA** [comment]
@MarcelCaron-NOAA The headline job has finished; please review the output. I did not notice any issues in the log.
### **jevs_cam_headline_plots**
**Log**: /lfs/h2/emc/vpppg/noscrub/shannon.shields/pr761test/EVS/dev/drivers/scripts/plots/cam/jevs_cam_headline_plots.o46184565
**DATA**: /lfs/h2/emc/stmp/shannon.shields/evs_test/prod/tmp/jevs_cam_headline_plots.46184565.dbqs01
**Plots**: /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/headline.20250616
Here is the size of the tar file:
`-rw-r--r-- 1 shannon.shields emc 5.1M Jun 17 15:29 evs.plots.cam.atmos.headline.v20250616.tar`

**Comment by MarcelCaron-NOAA** [comment]
@ShannonShields-NOAA Agreed, the job completed cleanly and output looks normal.  Attaching an image from that job which has the validation type name in the title.
add_obs_name_sample2

✔️ jevs_cam_headline_plots

**Comment by ShannonShields-NOAA** [comment]
@MarcelCaron-NOAA Great, thanks for sharing the example image! The rest of the jobs are done.
### **jevs_cam_snowfall_plots**
**Log**: /lfs/h2/emc/vpppg/noscrub/shannon.shields/pr761test/EVS/dev/drivers/scripts/plots/cam/jevs_cam_snowfall_plots.o46185412
Runtime:
```
ctime = Tue Jun 17 15:27:02 2025
mtime = Tue Jun 17 15:50:18 2025 (~23 mins)
Resource_List.walltime = 00:35:00
```
**DATA**: /lfs/h2/emc/stmp/shannon.shields/evs_test/prod/tmp/jevs_cam_snowfall_plots.46185412.dbqs01
**Plots**: /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/atmos.20250131
Tar file size:
`-rw-r--r-- 1 shannon.shields emc 574M Jun 17 15:50 evs.plots.cam.atmos.snowfall.v20250131.tar`
I noticed something that was previously noted in one of Perry's PRs. The plot output directory structure for snowfall looks like:
_/lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/atmos.20250131/**snowfall**/restart/**precip**_
Should both bolded directories be called snowfall to be consistent and avoid any potential confusion with your precip job?

### **jevs_cam_precip_plots_last31days**
**Log**: /lfs/h2/emc/vpppg/noscrub/shannon.shields/pr761test/EVS/dev/drivers/scripts/plots/cam/jevs_cam_precip_plots_last31days.o46184847
Runtime:
```
ctime = Tue Jun 17 15:24:08 2025
mtime = Tue Jun 17 15:58:26 2025 (~34 mins)
Resource_List.walltime = 00:45:00
```
I did notice a WARNING:
`06/17 15:33:20.150 (cam_plots_precip_spatial_map.py:192) WARNING: /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v2.0/stats/cam/atmos.20250615/namnest/precip/spatial_maps/namnest.init20250614.t12z.f024.a24h.conus.nc does not exist, not making plot`
Is this expected?
**DATA**: /lfs/h2/emc/stmp/shannon.shields/evs_test/prod/tmp/jevs_cam_precip_plots_last31days.46184847.dbqs01
**Plots**: /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/atmos.20250616
Tar file size:
`-rw-r--r-- 1 shannon.shields emc 345M Jun 17 15:58 evs.plots.cam.atmos.precip.last31days.v20250616.tar`

**Comment by MarcelCaron-NOAA** [comment]
@ShannonShields-NOAA The jobs completed cleanly and output files look normal.  

> I noticed something that was previously noted in one of Perry's PRs. The plot output directory structure for snowfall looks like:
> /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/atmos.20250131/snowfall/restart/precip
> Should both bolded directories be called snowfall to be consistent and avoid any potential confusion with your precip job?

I agree, it's confusing that snowfall restart plots are being stored in a "precip" subdirectory, which is the name for a different verif_case.  Do you mind rerunning that snowfall job with the recent change?

> I did notice a WARNING:
> 06/17 15:33:20.150 (cam_plots_precip_spatial_map.py:192) WARNING: /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v2.0/stats/cam/atmos.20250615/namnest/precip/spatial_maps/namnest.init20250614.t12z.f024.a24h.conus.nc does not exist, not making plot
> Is this expected?

This is expected.  This job usually runs later in the day because it relies on the output from a stats job that completes first.

So:
✔️ jevs_cam_precip_plots_last31days
🏁 jevs_cam_snowfall_plots

**Comment by ShannonShields-NOAA** [comment]
@MarcelCaron-NOAA Thank you for making the directory edit and explanation for the WARNING. I will rerun the snowfall job shortly. Btw, I love the check and flag icons you are using to mark progress; very organized!

**Comment by ShannonShields-NOAA** [comment]
@MarcelCaron-NOAA I have kicked off the snowfall job again and will let you know once it is done. Would you mind sharing an example precip graphic to show the obs source name in the title? Thanks!

**Comment by MarcelCaron-NOAA** [comment]
Sure!  Here are two images from the precip job:
add_obs_name_sample3
add_obs_name_sample

**Comment by ShannonShields-NOAA** [comment]
@MarcelCaron-NOAA Sorry for the delay; I realized I had to clear some directories. The snowfall job rerun has completed. Please review the output once more.
### **jevs_cam_snowfall_plots**
**Log**: /lfs/h2/emc/vpppg/noscrub/shannon.shields/pr761test/EVS/dev/drivers/scripts/plots/cam/jevs_cam_snowfall_plots.o46209841
**DATA**: /lfs/h2/emc/stmp/shannon.shields/evs_test/prod/tmp/jevs_cam_snowfall_plots.46209841.dbqs01
**Plots**: /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/atmos.20250131
Tar file size is the same as before:
`-rw-r--r-- 1 shannon.shields emc 574M Jun 17 18:40 evs.plots.cam.atmos.snowfall.v20250131.tar`
The directory structure is now consistent too: /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/atmos.20250131/**snowfall/restart/snowfall**

**Comment by MarcelCaron-NOAA** [comment]
@ShannonShields-NOAA No problem!  **jevs_cam_snowfall_plots** completed cleanly, output looks normal, the restart subdirectory reflects the recent name change to "snowfall", and plots include the title change (sample below):
add_obs_name_sample4

✔️ jevs_cam_snowfall_plots

👍 No other concerns from me about any of the test jobs for this PR!

File: ush/cam/cam_create_output_dirs.py [hunk 1]
```diff
@@ -678,7 +678,7 @@
             COMOUTplots, VERIF_CASE
         ))
         for plot_group in [
-                'cape', 'ceil_vis', 'precip', 'sfc_upper'
```
Comment (ShannonShields-NOAA) [inline ush/cam/cam_create_output_dirs.py:681]: @MarcelCaron-NOAA I was reviewing the script changes and I'm wondering if I need to test the headline job again? It looks like these were empty directories except for sfc_upper, so I'm assuming all restart plots are stored under sfc_upper?

Comment (ShannonShields-NOAA) [reply ush/cam/cam_create_output_dirs.py:681]: It can't hurt, so I am quickly re-running the headline job to confirm these restart directory changes before I leave for the day.

**Comment by ShannonShields-NOAA** [comment]
@MarcelCaron-NOAA The headline rerun job finished yesterday; please review the output. I did not notice any issues in the log.
### **jevs_cam_headline_plots**
**Log**: /lfs/h2/emc/vpppg/noscrub/shannon.shields/pr761test/EVS/dev/drivers/scripts/plots/cam/jevs_cam_headline_plots.o46221163
**DATA**: /lfs/h2/emc/stmp/shannon.shields/evs_test/prod/tmp/jevs_cam_headline_plots.46221163.dbqs01
**Plots**: /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/headline.20250616
Here is the size of the tar file (confirmed same size as before):
`-rw-r--r-- 1 shannon.shields emc 5.1M Jun 17 19:32 evs.plots.cam.atmos.headline.v20250616.tar`
I can also confirm the restart directory changes mentioned above:
_/lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/cam/headline.20250616/headline/restart/**sfc_upper**/last90days_

**Comment by MarcelCaron-NOAA** [comment]
@ShannonShields-NOAA I agree, the logfile is clean, output looks normal, and empty directories have been removed from restart.  No concerns from me!

**Review by ShannonShields-NOAA — APPROVED** [review]
@MarcelCaron-NOAA and @AndrewBenjamin-NOAA I have reviewed and confirmed the script changes (add obs name to plot title, change to subprocess.run, and adjustment of restart directories) work as expected. I approve this PR to be merged.

**Review by AndrewBenjamin-NOAA — APPROVED** [review]
I reviewed these changes and approve this PR

**Comment by ShannonShields-NOAA** [comment]
Thanks @AndrewBenjamin-NOAA ! @MarcelCaron-NOAA Can you please confirm your branch is up-to-date? It looks like it.

**Comment by MarcelCaron-NOAA** [comment]
> Thanks @AndrewBenjamin-NOAA ! @MarcelCaron-NOAA Can you please confirm your branch is up-to-date? It looks like it.

Yes, I checked and it was up to date.  Thanks @ShannonShields-NOAA!
